### PR TITLE
Feature/request error response

### PR DESCRIPTION
**Pull request by ZharaOo**

### CODE REVIEW

File: Sources/Extensions/Alamofire/SessionManager+Extensions.swift [hunk 1]
```diff
@@ -64,7 +68,43 @@ public extension Reactive where Base: SessionManager {
...
 
 }
+
+private extension ObservableType {
+    func catchErrorAndWrap(with request: DataRequest? = nil) -> Observable<E> {
+        return catchError {
+            let resultError: RequestError
+            let response = request?.delegate.data
+
+            switch $0 {
```
Comment (scoreyou) [inline Sources/Extensions/Alamofire/SessionManager+Extensions.swift]: name arguments please

File: Sources/Extensions/Alamofire/SessionManager+Extensions.swift [hunk 2]
```diff
@@ -64,7 +68,43 @@ public extension Reactive where Base: SessionManager {
...
+                switch urlError.code {
+                case .notConnectedToInternet:
+                    resultError = .noConnection
+                default:
+                    resultError = .network(error: urlError, response: response)
+                }
+            case let afError as AFError:
+                switch afError {
+                case .responseSerializationFailed, .responseValidationFailed:
+                    resultError = .invalidResponse(error: afError, response: response ?? Data())
```
Comment (scoreyou) [inline Sources/Extensions/Alamofire/SessionManager+Extensions.swift]: `?? Data()` seems strange

File: Sources/Extensions/Alamofire/DataRequest+Extensions.swift [hunk 3]
```diff
@@ -60,26 +60,6 @@ public extension Reactive where Base: DataRequest {
     private func response(onQueue queue: DispatchQueue) -> Observable<(HTTPURLResponse, Data)> {
         return responseData()
             .observeOn(SerialDispatchQueueScheduler(queue: queue, internalSerialQueueName: queue.label))
```
Comment (petropavel13) [inline Sources/Extensions/Alamofire/DataRequest+Extensions.swift]: replace with single line
```swift
responseResult(queue: queue, responseSerializer: DataRequest.dataResponseSerializer())
```

File: Sources/Extensions/Alamofire/SessionManager+Extensions.swift [hunk 4]
```diff
@@ -64,7 +68,43 @@ public extension Reactive where Base: SessionManager {
...
+                $0.rx.observableApiResponse(mappingQueue: self.base.mappingQueue, decoder: decoder)
+                    .catchErrorAndWrap(with: $0)
+            }
+            .catchErrorAndWrap()
     }
 
 }
+
+private extension ObservableType {
+    func catchErrorAndWrap(with request: DataRequest? = nil) -> Observable<E> {
```
Comment (petropavel13) [inline Sources/Extensions/Alamofire/SessionManager+Extensions.swift]: maybe rename to `catchAsRequestError` to make it more clear?

Comment (petropavel13) [reply Sources/Extensions/Alamofire/SessionManager+Extensions.swift]: or `rethrowAsRequestError()`

**Pull request merged**